src/glue.c

```diff
@@ -561,6 +563,33 @@ void genhelpers(Module *m, bool iscomdat)
 }
 
 /**************************************
+ * Bugzilla 14828: If some template instantiations require
+ * non-root module helpers, they should be placed in
+ * separated object file.
+ */
+void genHelpersObjFile(Library *library, Module *m)
+{
+    assert(!m->isRoot());
+
+    lastmname = m->srcfile->toChars();
+
+    obj_start(lastmname);
+
+    OutBuffer idbuf;
+    idbuf.printf("%s", m->ident->toChars());
+    char *idstr = idbuf.peekString();
+
+    idbuf.data = NULL;
+    Identifier *id = Identifier::create(idstr, TOKidentifier);
```

Comment:
Should prolly follow the naming scheme used here.
https://github.com/D-Programming-Language/dmd/blob/0f3850d2c620867460a4afcc23ae543f9163326f/src/glue.c#L121